Add note about potentially erroneous gcdumps (#28907)
dotnet/docs: docs/core/diagnostics/dotnet-gcdump.md

- COM and static types aren't in the GC dump.

>    Prior to .NET Core 3.1-preview2, there was an issue where static and COM types weren't sent when the GC dump was invoked via EventPipe. This has been fixed in .NET Core 3.1-preview2.

Reviewer: ```suggestion
   Prior to .NET Core 3.1, there was an issue where static and COM types weren't sent when the GC dump was invoked via EventPipe. This has been fixed in .NET Core 3.1.
```

> - `dotnet-gcdump` is unable to generate a `.gcdump` file due to missing information, e.g., `[Error] Exception during gcdump: System.ApplicationException: ETL file shows the start of a heap dump but not its completion.`, or my gcdump doesn't include the entire heap.

Reviewer: ```suggestion
- `dotnet-gcdump` is unable to generate a `.gcdump` file due to missing information, for example, **[Error] Exception during gcdump: System.ApplicationException: ETL file shows the start of a heap dump but not its completion.**. Or, the `.gcdump` file doesn't include the entire heap.
```

>    `dotnet-gcdump` works by collecting a trace of events emitted by the Garbage Collector during an induced Gen 2 GC. If the heap is sufficiently large or there is not enough memory to scale the eventing buffers, then there is the possibility of dropping the events required to reconstruct the heap graph from the trace. In this case, it is recommended to collect a dump of the process for diagnosing issues with the heap.

Reviewer: ```suggestion
   `dotnet-gcdump` works by collecting a trace of events emitted by the garbage collector during an induced generation 2 collection. If the heap is sufficiently large, or there isn't enough memory to scale the eventing buffers, then the events required to reconstruct the heap graph from the trace may be dropped. In this case, to diagnose issues with the heap, it's recommended to collect a dump of the process.
```
